docs/rep/node-agent/README.md

```diff
@@ -0,0 +1,99 @@
+# RIK - Node Agent Architecture (Riklet)
+
+This document provides a technical reflection about our component, the **node agent**.
+
+Our component will be installed on each node of a cluster and it will be composed of 4 sub-components :
+
+- **Node Manager**
+- **Node Runtime Memory**
+- **Node Proxy**
+- **Node Monitor**
+
+We will describe these components in more details throughout this document. Here is a diagram to illustrate how these components works together :
+
+![High-level view of Rikelet](assets/node-schema.png)
+
+## 1) Node Manager
+
+The Node Manager is the main sub-component of the Riklet. It should be able to :
+
+- Interprete Scheduler instructions
+- Run / Stop an application
```

Comment:
Could we define `rik` workload/pod/application lifecycle a little further?

Let's assume `rik` unit of execution are pods, for some definition of a pod:

- In which states can a pod be? (scheduled, started, running, stopped, paused, finished, etc). `riklet` should monitor each of those state transitions, and should make the cluster controller aware of them.
- Which of those states will `riklet` manage? Can it do something else than starting, stopping and killing a pod?

Reply:
These questions will be answered in the next PR